### ISSUE TITLE
Indentation regression with inline lambda in function call

### ISSUE THREAD

**Issue by Gama11** [pinterest/ktlint#1202]
Note: this is a regression compared to 0.41.0 that is still present in the 0.42.1 patch release where some indentation bugs were already fixed.

`ktlint -F` turns the following:

```kt
class Foo {
    fun bar() {
        val foo = bar.associateBy({ item -> item.toString() }, ::someFunction).toMap()
    }
}
```

into

```kt
class Foo {
    fun bar() {
        val foo = bar.associateBy({ item -> item.toString() }, ::someFunction).toMap()
        }
    }
```

**Comment by shashachu** [comment]
@Gama11 thanks for the report. we have a few rules than can touch indentation (likely part of the problem). Can you run `ktlint --verbose` to verify which of the rules is reformatting it?

**Comment by Gama11** [comment]
Sure - I assume `indent` is the name of the rule:

>Unexpected indentation (4) (should be 8) (indent)
>Unexpected indentation (0) (should be 4) (indent)
>Unexpected indentation (0) (should be 4) (indent)
